### ISSUE TITLE
How to make the right click behave as if no orbit controls exist?

### ISSUE THREAD

**Issue by youssefsharief** [yomotsu/camera-controls#361]
Question in title

**Comment by yomotsu** [comment]
Sorry, I don't understand your situation.
I think you don't need any care of it if you don't use it...

**Comment by youssefsharief** [comment]
@yomotsu 
The issue is that typescript enforces me to add the "right" prop to the "mouseButtons" object. Even when I set the right prop to "CameraControls.ACTION.NONE", I am still not able to use right click appropriately


```
     mouseButtons={{
        left: CameraControls.ACTION.TRUCK,
        right: CameraControls.ACTION.NONE,
        middle: CameraControls.ACTION.ZOOM,
        wheel: CameraControls.ACTION.ZOOM,
      }}
```

**Comment by yomotsu** [comment]
how about calling `cameraControls.dispose()` to detach all events of cameraControls instance?

**Comment by youssefsharief** [comment]
If I called "cameraControls.dispose()", would I need to manually add an event listener to "left", "middle", and "wheel"?

**Comment by yomotsu** [comment]
Check the documentation for more details!
https://github.com/yomotsu/camera-controls#dispose
https://github.com/yomotsu/camera-controls#disconnect

**Comment by youssefsharief** [comment]
I checked the documentation and I still believe that "CameraControls.ACTION.NONE" for the right click button should allow default action

**Comment by yomotsu** [comment]
Let me get this straight.
What you mean is: you would like to enable the **context menu** if you set `mouseButtons.right = CameraControls.ACTION.NONE`?

**Comment by yomotsu** [comment]
I tested the below, but it cause some problems.
```
 const onContextMenu = ( event: Event ): void => { 
  
 	if ( ! this._enabled || mouseButtons.right = CameraControls.ACTION.NONE ) return; 
  
 	event.preventDefault(); 
  
 };
```

contextmenu event is fired right after mouse down event, and contextmenu will break mouse dragging.
thus, we can't change the above code.

**Comment by youssefsharief** [comment]
> What you mean is: you would like to enable the context menu if you set mouseButtons.right = CameraControls.ACTION.NONE?

Yes

**Comment by yomotsu** [comment]
I've made a PR for it https://github.com/yomotsu/camera-controls/pull/362

**Comment by youssefsharief** [comment]
@yomotsu thank you so much

**Comment by yomotsu** [comment]
available with v2.2.1!
let me close this issue as complicated.
